charms/slurmd/src/utils/rdma.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Manage RDMA packages on compute nodes."""
+
+import logging
+from pathlib import Path
+
+import charms.operator_libs_linux.v0.apt as apt
+
+_logger = logging.getLogger(__name__)
+
+
+class RDMAOpsError(Exception):
+    """Exception raised when an RDMA package installation operation failed."""
+
+    @property
+    def message(self) -> str:
+        """Return message passed as argument to exception."""
+        return self.args[0]
+
+
+def _install_rdma(install_packages: list) -> None:
```

Comment:
[nit]: it's preferred to use an abstract type from `collections.abc` to annotate arguments rather than use [`list`/`typing.List`](https://docs.python.org/3/library/typing.html#typing.List). Here it would probably make the most sense to use `Iterable` since we're just iterating over the list of packages to install

```suggestion
def _install_rdma(install_packages: Iterable[str]) -> None:
```

Note that in another review comment of mine, I don't think we need to pass an argument here. We should just declare the package set directly within the function.

Reply:
Noted and fixed by this function no longer taking an argument.